### PR TITLE
Ignore "release: not found" error on helm_resource destroy

### LINKED ISSUE #1488: Handle "release not found" errors thrown on uninstall / equivalent to `helm uninstall --ignore-not-found`
### Description

We are frequently hitting an issue with `helm` TF provider failing to destroy a `helm_release` resource when a K8s cluster (specifically GKE 1.28) throws an `Internal Server Error` and `terraform destroy` than fails with:
```
Error: uninstallation completed with 1 error(s): an error on the server (\\\"Internal Server Error: \\\\\\\"/api/v1/namespaces/ingress-nginx/services/ingress-nginx-controller-admission\\\\\\\": the server is currently unable to handle the request\\\") has prevented the request from succeeding (get services ingress-nginx-controller-admission)\\n\\n\"
```

Retrying the destroy operation then fails with `release not found` uninstallation error, e.g.:
```
Error: uninstall: Release not loaded: ingress-nginx: release: not found"
```

as the `helm_release` resource remains in the TF state, but its resources are already deleted from the K8s cluster.

In our case, we are destroying whole cluster so we want to ignore such errors. Helm CLI already provides a solution for this specific situation: `helm uninstall --ignore-not-found`:
```
$ helm uninstall --help |grep ignore-not-found
      --ignore-not-found     Treat "release not found" as a successful uninstall
```

### Potential Terraform Configuration

https://github.com/hashicorp/terraform-provider-helm/pull/1487

Add `ignore_not_found` boolean attribute (defaulting to `false`) to the `helm_release` resource, enabling to ignore the error on uninstall.

```hcl
resource "helm_release" "ingress_nginx" {
  name              = local.ingress_nginx
  chart             = local.ingress_nginx
  repository        = "https://kubernetes.github.io/ingress-nginx"
  version           = var.ingress_nginx_chart_version
  namespace         = local.ingress_nginx_namespace
  atomic            = true
  wait              = true
  wait_for_jobs     = true
  skip_crds         = true
  lint              = true
  cleanup_on_fail   = true
  dependency_update = false
  create_namespace  = false
  ignore_not_found  = true
}
```

### References

- https://github.com/hashicorp/terraform-provider-helm/pull/1487

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by sbocinec**
### Description

Ignore "release: not found" error on helm_resource destroy to address [GH-1488](https://github.com/hashicorp/terraform-provider-helm/issues/1488)

### Acceptance tests
- [ ] Have you added an acceptance test for the functionality being added?

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-helm/blob/main/CHANGELOG.md):

```release-note
resource/helm_release: ignore "release not found" destroy error, addressing [GH-1488](https://github.com/hashicorp/terraform-provider-helm/issues/1488)
```
### References
- https://github.com/hashicorp/terraform-provider-helm/issues/1488
- https://helm.sh/docs/helm/helm_uninstall/

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

Closes #1488

### CODE REVIEW

File: helm/resource_release_test.go [hunk 1]
```diff
@@ -1758,6 +1758,45 @@ func TestAccResourceRelease_delete_regression(t *testing.T) {
 	})
 }
 
+func TestAccResourceRelease_destroy_ignoreReleaseNotFoundError(t *testing.T) {
```
Comment (appilon) [inline helm/resource_release_test.go]: It's not really something we tend to test for let's remove the acceptance test and just go with the codechange. We will run the full suite of tests to ensure no regressions.

**Review by appilon — CHANGES_REQUESTED** [review]

File: helm/resource_release_test.go (hunk 1 above)
Comment (sbocinec) [reply helm/resource_release_test.go]: okay, I have removed it

**Comment by appilon** [comment]
I will add the changelog entry for you and get this merged

**Review by appilon — APPROVED** [review]

**Comment by sbocinec** [comment]
@appilon I think, we must revert this PR - I have just tested it once it was released yesterday, and it panics https://github.com/hashicorp/terraform-provider-helm/issues/1500

**Pull request merged**